storage/storage.cpp

```diff
@@ -1145,4 +1145,19 @@ void Storage::DoClickOnDownloadMap(TCountryId const & countryId)
   if (m_downloadMapOnTheMap)
     m_downloadMapOnTheMap(countryId);
 }
+
+void Storage::ForEachInSubtree(TCountryId const & root, TForEachFunction const & toDo) const
```

Comment:
It looks like 

void ForEachInSubtree(TCountryId const & root, TForEachFunction && toDo) const;
and 
void ForEachInSubtree(TCountryId const & root, TForEachFunction const & toDo) const;
are almost equal, except case:
  TForEachFunction myToDo = ...;
  ForEachInSubtree(myToDo);
this case wont compile in first case because param is &&

It looks like 
void ForEachInSubtree(TCountryId const & root, TForEachFunction const & toDo) const;
is more preferable.